screens/TournamentDetails.tsx

```diff
@@ -0,0 +1,172 @@
+import { Route } from "@react-navigation/native";
+import React, { useContext, useEffect, useState } from "react";
+import { Modal, StyleSheet } from "react-native";
+import { TouchableOpacity } from "react-native-gesture-handler";
+import { getTournament, addBracket } from "../api/Requests";
+import Spinner from "../components/Spinner";
+
+import { Text, View, LinkButton, TextInput } from "../components/Themed";
+import { AuthUserContext } from "../contexts/AuthContext";
+import { Tournament, Bracket } from "../models";
+import { ScreenWithNavigation } from "../types";
+
+type DetailScreen = ScreenWithNavigation & { route: Route<any> }
+
+export default function TournamentDetailsScreen({ navigation, route }: DetailScreen) {
+
+    const { user } = useContext(AuthUserContext);
+    const { id: tournamentId } = route.params as { id: string };
+
+    // TODO: Update data and send to API when done
+    const [tournament, setTournament] = useState<Tournament | undefined>();
+    const [loading, setLoading] = useState(true);
+
+    const [bracket, setBracket] = useState<Bracket>({name: ''})
+
+    // For Adding brackets
+    const [showAddBracket, setShowAddBracketModal] = useState(false);
+
+    const fetchTournamentDetails = async () => {
+        const data = await getTournament(tournamentId, user);
+        setTournament(data);
+        setLoading(false)
+    }
+
+    // On screen load, fetch the tournament details
+    useEffect(() => {
+        fetchTournamentDetails();
+    }, []);
+
+    const submitAddBracket = async () => {
+        setLoading(true);
+        await addBracket(bracket, tournamentId, user);
+        setBracket({name: ''});
+        // Refresh w/ new bracket
+        fetchTournamentDetails();
+    }
+
+    if (loading || tournament === undefined) {
+        return (
+            <Spinner />
+        );
+    }
+
+    const renderAddBracketModal = () => {
+        return (
+            <Modal
+                animationType="slide"
+                transparent={true}
+                visible={showAddBracket}
```

Comment:
So if this is visible, you don't see the stuff below it, and if it's invisible you see the stuff below? I should refactor the other stuff into Modals then

(edit: by below I mean when renderAddBracketModal is called)

Reply:
Well ideally it would still show the rest of the screen behind it, but like with a shadow, not sure why this background seems to be opaque. Will look into it